dot/parachain/availability-distribution/availability_distribution.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2025 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package availabilitydistribution
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+)
+
+var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-availability-distribution"))
+
+type AvailabilityDistribution struct {
+	subSystemToOverseer chan<- any
+	net                 Network
+	blockState          BlockState
+}
+
+var _ parachaintypes.Subsystem = (*AvailabilityDistribution)(nil)
+
+type Network interface {
+	RegisterRequestHandler(subprotocolID protocol.ID, handler network.RequestHandler)
```

Comment:
I might be missing something, but why can the subsystem register a handler? I mean, in what specific case the subsystem will register a request handler for a specific network protocol id?

Reply:
The availability distribution subsystem is responsible for fetching erasure chunks and PoVs as well as responding to those requests from other nodes.

So on startup the subsystem will register handlers for `req_chunk/1` (and/or `req_chunk/2`) and `req_pov/1`.

Reply:
@haikoschol Do we need both req_chunk/1 and req_chunk/2?

Reply:
> @haikoschol Do we need both req_chunk/1 and req_chunk/2?

Probably not. I've only covered v2 in #4598.